src/actions/AddPeriodicControllers.C

```diff
@@ -138,77 +143,93 @@ AddPeriodicControllers::act()
+      if (_cycles_between_controls == 1.0)
       {
         for (MooseIndex(_enable_end) j = 0; j < _enable_end.size(); ++j)
         {
           _enable_end_start_time_index[j] =
-              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+              (_start_time + _period * _cycles_per_controls) +
+              (_cycles_between_controls + _cycles_per_controls) * _period * i + (_period * 0.0001);
           _enable_end_end_time_index[j] =
-              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0002);
+              (_start_time + _period * _cycles_per_controls) +
+              (_cycles_between_controls + _cycles_per_controls) * _period * i + (_period * 0.0002);
         }
       }
       else
       {
         for (MooseIndex(_enable_end) j = 0; j < _enable_end.size(); ++j)
         {
           _enable_end_start_time_index[j] =
-              (_start_time + _period) + _cycles_per_controls * _period * i;
+              (_start_time + _period * _cycles_per_controls) +
+              (_cycles_between_controls + _cycles_per_controls) * _period * i;
           _enable_end_end_time_index[j] =
-              (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+              (_start_time + _period * _cycles_per_controls) +
+              (_cycles_between_controls + _cycles_per_controls) * _period * i + (_period * 0.0001);
         }
       }
 
       for (MooseIndex(_disable_start) j = 0; j < _disable_start.size(); ++j)
       {
-        _disable_start_start_time_index[j] = _start_time + _cycles_per_controls * _period * i;
+        _disable_start_start_time_index[j] =
+            _start_time + (_cycles_between_controls + _cycles_per_controls) * _period * i;
         _disable_start_end_time_index[j] =
-            _start_time + _cycles_per_controls * _period * i + (_period * 0.0001);
+            _start_time + (_cycles_between_controls + _cycles_per_controls) * _period * i +
+            (_period * 0.0001);
       }
 
       for (MooseIndex(_disable_during) j = 0; j < _disable_during.size(); ++j)
       {
-        _disable_during_start_time_index[j] = _start_time + _cycles_per_controls * _period * i;
+        _disable_during_start_time_index[j] =
+            _start_time + (_cycles_between_controls + _cycles_per_controls) * _period * i;
         _disable_during_end_time_index[j] =
-            (_start_time + _period) + _cycles_per_controls * _period * i + (_period * 0.0001);
+            (_start_time + _period * _cycles_per_controls) +
+            (_cycles_between_controls + _cycles_per_controls) * _period * i + (_period * 0.0001);
       }
 
       // If control cycles are back to back, then disable_start and disable_end times would be the
       // same, To avoid this, if cycles are back to back, then the disable_end starts at +
       // 0.0001*period
-      if (_cycles_per_controls == 1.0)
+      if (_cycles_between_controls == 1.0)
```

Comment:
We really shouldn't be doing this sort of float equality comparison. we should be using the absolute fuzzy equals functionality for this sort of comparison.